### ISSUE TITLE
Attention scaling is never used

### ISSUE THREAD

**Issue by francesco-bertolotti** [pytorch/torchtitan#2170]
In the Qwen3 model implementation, attention scaling is defined but never actually applied.

Specifically, in the following code, the scaling factor is mentioned but not used:
https://github.com/pytorch/torchtitan/blob/1bd2548b14da014b1ec560830f8bdefb6ca568f4/torchtitan/models/qwen3/model/model.py#L176

I suspect this may be a byproduct of using the PyTorch 2.1+ Flash Attention backend, where scaling is handled internally.

Is this behavior expected? If so, are there any recommended workarounds or configuration options to explicitly control attention scaling?

NB: Deepseek implementation does not suffer from this but it does not support the flash attention backend

**Comment by tianyu-l** [comment]
cc @wwwjn

**Comment by wwwjn** [comment]
Thanks for asking and nice catch! The scaling was introduced in PR: https://github.com/pytorch/torchtitan/pull/1704/changes#diff-b325355aeabc98b656c761ee8faef22c0022c7cb6e11afaed482cf4231d9feccL212 and we were applying it to SDPA attention.  In later refactors, this field is dropped.

For sdpa, we need to fix and passing scaling during forward - I will fix this and create a PR

For varlen attention / Flash attention - cc @drisspg  Do you know how we handle scale?

For FlexAttention, we need to fix and passing scale here: https://github.com/pytorch/torchtitan/blob/1bd2548b14da014b1ec560830f8bdefb6ca568f4/torchtitan/models/attention.py#L135

**Comment by drisspg** [comment]
@liangel-02 can you add softmax scaling to varlen_api in pt and then make the update here

**Comment by wwwjn** [comment]
https://github.com/pytorch/torchtitan/pull/2173 Here's for Flex and sdpa attention
